cmd/store/export.go

```diff
@@ -0,0 +1,185 @@
+	if err != nil {
+		return nil, fmt.Errorf("unable to get model dsl: %w", err)
+	}
+
+	// get the tuples
+	maxPages := int(math.Ceil(float64(maxTupleCount) / float64(tuple.DefaultReadPageSize)))
+
+	rawTuples, err := tuple.Read(fgaClient, &client.ClientReadRequest{}, maxPages)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read tuples: %w", err)
+	}
+
+	tuples := rawTuples.GetTuples()
+	maxTuplesInOutput := int(math.Min(float64(len(tuples)), float64(maxTupleCount)))
+	outputTuples := make([]client.ClientContextualTupleKey, 0, maxTuplesInOutput)
+
+	for _, t := range tuples[:maxTuplesInOutput] {
+		outputTuples = append(outputTuples, t.GetKey())
+	}
+
+	// get the assertions
+	assertionOptions := client.ClientReadAssertionsOptions{
+		AuthorizationModelId: authModel.ID,
+	}
+
+	assertionResponse, err := fgaClient.ReadAssertions(context.Background()).Options(assertionOptions).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("unable to read assertions: %w", err)
+	}
+
+	assertions := assertionResponse.GetAssertions()
+	modelChecks := map[string]storetest.ModelTestCheck{}
+
+	for _, assertion := range assertions {
+		key := fmt.Sprintf("%s|%s", assertion.TupleKey.User, assertion.TupleKey.Object)
+		_, exists := modelChecks[key]
+
+		if !exists {
+			modelChecks[key] = storetest.ModelTestCheck{
+				User:       assertion.TupleKey.User,
+				Object:     assertion.TupleKey.Object,
+				Assertions: map[string]bool{},
+			}
+		}
+
+		modelChecks[key].Assertions[assertion.GetTupleKey().Relation] = assertion.Expectation
+	}
+
+	checks := []storetest.ModelTestCheck{}
+	for _, value := range modelChecks {
+		checks = append(checks, value)
+	}
+
+	storeData := &storetest.StoreData{
+		Name:   storeName,
+		Model:  *dsl,
+		Tuples: outputTuples,
+		Tests: []storetest.ModelTest{
+			{
+				Name:  "Tests",
+				Check: checks,
+			},
+		},
+	}
+
+	return storeData, nil
+}
+
+// exportCmd represents the export store command.
+var exportCmd = &cobra.Command{
+	Use:     "export",
+	Short:   "Export store data",
+	Long:    `Export a store to YAML`,
+	Example: "fga store export",
+	RunE: func(cmd *cobra.Command, _ []string) error {
+		clientConfig := cmdutils.GetClientConfig(cmd)
+
+		fgaClient, err := clientConfig.GetFgaClient()
+		if err != nil {
+			return fmt.Errorf("failed to initialize FGA Client due to %w", err)
+		}
+
+		maxTupleCount, _ := cmd.Flags().GetUint("max-tuples")
+		storeData, err := buildStoreData(clientConfig, fgaClient, maxTupleCount)
+		if err != nil {
+			return fmt.Errorf("failed to export store: %w", err)
+		}
+
+		if storeData != nil {
+			storeYaml, err := yaml.Marshal(storeData)
+			if err != nil {
+				return fmt.Errorf("unable to marshal storedata yaml: %w", err)
+			}
+
+			fileName, _ := cmd.Flags().GetString("output-file")
+			if fileName == "" {
+				fmt.Println(string(storeYaml))
+
+				return nil
+			}
+
+			err = os.WriteFile(fileName, storeYaml, 0o600) //nolint:gomnd
```

Comment:
Given that `WriteFile` will overwrite any existing files should we perform a check if the file exists and then show a confirmation prompt to avoid any destructive changes?

Reply:
Added in https://github.com/openfga/cli/pull/306/commits/dc82f9166984daf314dd3ea669c6a63ec739a421